### PR TITLE
LPD-23257 FE | Support Regular and Organization Roles for Email notification Template 

### PR DESCRIPTION
**Pull request by igor-franca**
Related [ticket](https://liferay.atlassian.net/browse/LPD-23257)

### CODE REVIEW

File: modules/apps/notification/notification-web/src/main/resources/META-INF/resources/js/components/SettingsContainer/PrimaryRecipients.tsx [hunk 1]
```diff
@@ -49,12 +49,13 @@ export function PrimaryRecipient({
...
-				if (child.checked) {
-					newRecipients.push({['roleName']: child.value});
-				}
+			const itemsGroup = items as MultiSelectItem[];
+			itemsGroup.map((itemGroup) => {
+				itemGroup.children.forEach((child) => {
+					if (child.checked) {
+						newRecipients.push({['roleName']: child.value});
+					}
+				});
```
Comment (danvb112) [inline modules/apps/notification/notification-web/src/main/resources/META-INF/resources/js/components/SettingsContainer/PrimaryRecipients.tsx:58]: Hey @igor-franca, this line `52` is unnecessary, you can use `items` directly or rename it to `itemsGroup` in the function parameter.

Also, if you are not going to return anything, you don't need to use the `map` function, use `forEach` instead

File: modules/apps/notification/notification-web/src/main/resources/META-INF/resources/js/components/SettingsContainer/SecondaryRecipients.tsx [hunk 2]
```diff
@@ -42,12 +42,14 @@ export function SecondaryRecipient({
...
-			itemsGroup.children.forEach((child) => {
-				if (child.checked) {
-					newRecipients.push({['roleName']: child.value});
-				}
+			itemsGroup.map((itemGroup) => {
+				itemGroup.children.forEach((child) => {
+					if (child.checked) {
+						newRecipients.push({['roleName']: child.value});
+					}
+				});
```
Comment (danvb112) [inline modules/apps/notification/notification-web/src/main/resources/META-INF/resources/js/components/SettingsContainer/SecondaryRecipients.tsx:52]: Hey @igor-franca, same as the comment I made on `PrimaryRecipients`

File: modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/components/Select/MultipleSelect.tsx [hunk 3]
```diff
@@ -74,8 +74,7 @@ export function MultipleSelect({
 	useEffect(() => {
 		if (selectAllOption) {
 			let firstRender = false;
-
-			let notAllSelected: MultiSelectItemChild | undefined;
+			let allSelected = true;
 
 			(options as MultiSelectItem[]).forEach(({children}) => {
```
Comment (danvb112) [inline modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/components/Select/MultipleSelect.tsx:79]: Hey @igor-franca, we already have the type of this `option`, so we can remove this type declaration here

**Review by danvb112 — CHANGES_REQUESTED** [review]
Hey @igor-franca, good job on this PR, I have some suggestions :smile: 


Also, I would suggest you to rename the `rolesUtils.ts > rolesUtil.ts`

**Comment by igor-franca** [comment]
Hey Dan, I actually made some mistakes related to the naming of roles and acocunts. Thanks for highlighting this!

**Comment by danvb112** [comment]
ci:test:sf

**Review by danvb112 — APPROVED** [review]
LGTM! :tada:

**Comment by igor-franca** [comment]
ci:test:object

**Comment by igor-franca** [comment]
ci:test:relevant

**Comment by igor-franca** [comment]
After [comparing](https://testray.liferay.com/home/-/testray/runs/compare?formDate=1714044097505&testrayRunIdA=2563457640&testrayRunIdB=2563480590) the test results and running some tests locally, I was able to conclude that no error is related to the changes in this PR

**Comment by igor-franca** [comment]
ci:forward

**Comment by igor-franca** [comment]
ci:stop

**Comment by igor-franca** [comment]
ci:stop

**Comment by igor-franca** [comment]
ci:forward:force

**Pull request closed without merging**